src/v0/destinations/google_adwords_offline_conversions/utils.js

```diff
@@ -54,6 +65,48 @@ const validateDestinationConfig = ({ Config }) => {
   }
 };
 
+/**
+ * Determines whether to use batch fetching for conversion metadata.
+ *
+ * When enabled (true):
+ * - Conversion action IDs and custom variables are fetched in batches
+ *   during the transformation phase
+ * - Reduces API calls and improves performance
+ * - Data is cached and reused across events
+ *
+ * When disabled (false):
+ * - Legacy flow: fetches conversion data per-request in network handler
+ * - One API call per event (slower but battle-tested)
+ *
+ * @returns {boolean} true to enable batch fetching, false for legacy per-request flow
+ */
+const isBatchFetchingEnabled = () => process.env.GAOC_ENABLE_BATCH_FETCHING === 'true';
+
+const getHeader = (Config, metadata, passToken = false) => {
+  const { subAccount, loginCustomerId } = Config;
+  const response = {
+    Authorization: `Bearer ${getAccessToken(metadata, 'access_token')}`,
+    'Content-Type': 'application/json',
+  };
+
+  const useBatchFetching = isBatchFetchingEnabled();
+  const developerToken = getDeveloperToken();
+  if (developerToken && useBatchFetching && passToken) {
+    response['developer-token'] = developerToken;
+  }
```

Comment:
No need to invoke unless it is required.

```suggestion
  if (passToken) {
    const useBatchFetching = isBatchFetchingEnabled();
    const developerToken = getDeveloperToken();
    if (developerToken && useBatchFetching) {
      response['developer-token'] = developerToken;
    }
  }
```